pkg/controller/endpointslice/topologycache/sliceinfo_test.go

```diff
@@ -35,28 +36,36 @@ func TestGetTotalEndpoints(t *testing.T) {
 	}, {
 		name: "empty slice",
 		si: &SliceInfo{
-			ToCreate: []*discovery.EndpointSlice{sliceWithNEndpoints(0)},
+			ToCreate: []*discovery.EndpointSlice{sliceWithNEndpoints(0, 0)},
 		},
 		expectedTotal: 0,
 	}, {
 		name: "multiple slices",
 		si: &SliceInfo{
-			ToCreate: []*discovery.EndpointSlice{sliceWithNEndpoints(15), sliceWithNEndpoints(8)},
+			ToCreate: []*discovery.EndpointSlice{sliceWithNEndpoints(15, 0), sliceWithNEndpoints(8, 0)},
 		},
 		expectedTotal: 23,
 	}, {
 		name: "slices for all",
 		si: &SliceInfo{
-			ToCreate:  []*discovery.EndpointSlice{sliceWithNEndpoints(15), sliceWithNEndpoints(8)},
-			ToUpdate:  []*discovery.EndpointSlice{sliceWithNEndpoints(2)},
-			Unchanged: []*discovery.EndpointSlice{sliceWithNEndpoints(100), sliceWithNEndpoints(90)},
+			ToCreate:  []*discovery.EndpointSlice{sliceWithNEndpoints(15, 0), sliceWithNEndpoints(8, 0)},
+			ToUpdate:  []*discovery.EndpointSlice{sliceWithNEndpoints(2, 0)},
+			Unchanged: []*discovery.EndpointSlice{sliceWithNEndpoints(100, 0), sliceWithNEndpoints(90, 0)},
 		},
 		expectedTotal: 215,
+	}, {
+		name: "slices for all with some unready",
+		si: &SliceInfo{
+			ToCreate:  []*discovery.EndpointSlice{sliceWithNEndpoints(15, 3), sliceWithNEndpoints(5, 4)},
+			ToUpdate:  []*discovery.EndpointSlice{sliceWithNEndpoints(3, 8)},
+			Unchanged: []*discovery.EndpointSlice{sliceWithNEndpoints(98, 2), sliceWithNEndpoints(90, 6)},
+		},
+		expectedTotal: 211,
```

Comment:
so this test would still pass even if `sliceWithNEndpoints` didn't create the unready endpoints... maybe check both total endpoints and total unready endpoints?

Reply:
Updated, thanks!